### ISSUE TITLE
Implement lazy load

### ISSUE THREAD

**Issue by rickmcgeer** [rickmcgeer/sdtp-data-wiki#4]
At the moment, we load all the tables in the store immediately on startup, which will (when we have a lot of tables) lead to slow startup and it will also take a lot of memory.  A better idea is to load a table only in response to a request, and then swap it out after an idle-time period.

**Comment by rickmcgeer** [comment]
The basic tool we need was added to the sdtp package in 0.2.5 -- `ReloadableTable` which has the concrete subclasses `FileTable`, `HTTPTable`, `GCSTable`.  These permit the schema for a table to be stored with a pointer to another table (typically a `RowTable`) with the actual data, e.g.
```
{
    "type": "GCSTable",
    "schema": [{"name": "name", "type": "string"}, {"name": "age", "type": "number" }],
    "bucket": "mybucket",
    "blob": "foobar/sample_table.sdml"
}
```
Where `foobar/sample_table.sdml` is a standard `RowTable`:
```
{
    "type": "RowTable",
    "schema": [{"name": "name", "type": "string"}, {"name": "age", "type": "number" }],
    "rows":  [['"Bob", 22], ["Alice", 23]]
}
```
To take advantage of this, we should use two trees in the storage bucket: `gcstables` and `rowtables`, where the gcstables are stored as `gcstables/<user>/sample_table.sdml` and the rowtables are stored as `rowtables/<user>/sample_table.sdml`

To implement this, in `upload_file` in `main.py`, we add the code:
```
gcs_table_spec = {
     "schema":  table_dictionary["schema"],
     "type": "GCSTable",
     "bucket": BUCKET_NAME,
     "blob": f"rowtables/{table_dictionary['name']}",
     "name": f"gcstables/{table_dictionary['name']}"
}
```
Which creates the `GCSTable` as a dictionary.  We then replace the existing line:
```
sdtp_server_blueprint.table_server.add_sdtp_table_from_dictionary(table_dictionary["name"], table_dictionary["table"])
```
with
```
sdtp_server_blueprint.table_server.add_sdtp_table_from_dictionary(table_dictionary["name"], gcs_table_spec)
```
finally, we have to store both tables
After
```
bucket.upload_table(table_dictionary)
```
Add
```
bucket.upload_table(gcs_table_spec)
```

**Comment by rickmcgeer** [comment]
Implemented in 5990563402fc6c6c63b8b23a2f1b277a901ffbba, using the architecture described in the comments here.
